src/Components/Server/src/Builder/ServerComponentsEndpointOptions.cs

```diff
@@ -0,0 +1,46 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Connections;
+
+namespace Microsoft.AspNetCore.Components.Server;
+
+/// <summary>
+/// Options to configure interactive Server components.
+/// </summary>
+public class ServerComponentsEndpointOptions
```

Comment:
I've simplified this and left only the callback, since the plan is for it to be enabled by default, turning it off is a matter of setting the callback to null.